### PR TITLE
Fix 4949 settings rendering tweaks

### LINKED ISSUE #4949: Settings treeview rendering in High DPI
**Do you want to request a *feature* or report a *bug*?**
visual defect in hi-dpi
relates to #4775 

**What is the current behavior?**
* Treenodes look squashed
image

* The search box looks unaligned

**What is the expected behavior?**
* Treeview looks similar to other treeviews present in the app (e.g. LeftPanel, Revision File Tree)

### PR DESCRIPTION
**Pull request by RussKie**
* Render the treeview consistently with the other treeviews
* Align the search textbox with the treeview

Fixes #4949

 
Screenshots before and after (if PR changes UI):
before:
image

after:
image

What did I do to test the code and ensure quality:
- manual tests at 250%

### CODE REVIEW

File: GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs [hunk 1]
```diff
@@ -27,6 +28,10 @@ public SettingsTreeViewUserControl()
 
             _origTextBoxFont = textBoxFind.Font;
             SetFindPrompt(true);
+
+            treeView1.ImageList = new ImageList();
+            treeView1.ImageList.ImageSize = DpiUtil.Scale(new Size(16, 16)); // Scale ImageSize and images scale automatically
```
Comment (RussKie) [inline GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: @drewnoakes @Quppa I'm having a bit of a conundrum. 
If I don't scale the image size I don't get `>` icons the same size as in the other treeviews.
However now I get an empty space, absent any images...

Thoughts? Suggestions?

Comment (Quppa) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: You could try one of the workarounds listed [here](https://stackoverflow.com/a/31507872) to hide the images.

I thought that blank space to the right of the textbox was being used to show some sort of indicator after searching - has that changed?

Comment (RussKie) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: >  thought that blank space to the right of the textbox was being used to show some sort of indicator after searching - has that changed?

Indeed, it is... Thanks for pointing.

Comment (RussKie) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: > You could try one of the workarounds listed here to hide the images.

Thank you
I don't really like the idea of custom drawing the whole thing, especially in wake of dealing with scaling issues...
[A possible solution from a mailing list](http://www.tech-archive.net/Archive/DotNet/microsoft.public.dotnet.framework.windowsforms/2006-09/msg00322.html) looked promising but it only seems to work for hovered nodes...

image

It feels like a blank image is drawn anyway.

Comment (Quppa) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: Any chance setting StateImageList helps (one of the other solutions listed on StackOverflow)?

Agreed about avoiding custom draw if at all possible.

**Comment by RussKie** [comment]
I kind of tried it too, but it is late here and I gave up.
Something I thought could be a quick tweak turns into a major headache.

File: GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs (hunk 1 above)
Comment (drewnoakes) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: Agree that the [`StateImageList` approach](https://stackoverflow.com/a/31507872/24874) sounds promising. Did you try that one @RussKie?

Comment (RussKie) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: I've seen this comment, but I'm not sure how to put it to use...

Comment (drewnoakes) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: My interpretation is to replace `ImageList` with `StateImageList` and `ImageIndex` with `StateImageIndex`.

Comment (RussKie) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: I tried replacing `ImageList` with `StateImageList` and it doesn't work. Perhaps I'm missing something bleeding obvious....

It hurts, but I'm giving up. The proposal is not perfect, but it is more visually consistent with the rest. We can add icons for each entry later.

**Comment by RussKie** [comment]
The old:
image


The new:
image

**Comment by drewnoakes** [comment]
Definitely an improvement for high DPI users 👍 

As an aside, this is a great example of the kind of pain that goes away with WPF where you get complete control over the rendering of things such as a tree view. I don't want to derail another issue by mentioning WPF so opened #4972 for such discussion.

**Pull request merged**